sbin/build.sh

```diff
@@ -2094,7 +2094,9 @@ getTargetFileNameForComponent() {
     echo "${target_file_name}" | sed "s/-jdk/-${component}/"
   else
     # if no pattern is found, append the component name right before the extension.
-    echo "${target_file_name}" | sed -r "s/(.+)(\.tar\.gz|\.zip)/\1-${component}\2/"
+    echo "${target_file_name}" | sed \
```

Comment:
Worth adding a comment about using Solaris friendly sed?  I've seen a few cases where this sort of thing gets reverted by Linux focused folks :-)

Reply:
Isn't shell check able to to check posix? But I guess that is syntax only and not sed/grep and others...

Reply:
> Worth adding a comment about using Solaris friendly sed? I've seen a few cases where this sort of thing gets reverted by Linux focused folks :-)

I have a view on that ... If we think we really need to tell people explictly "Don't change this to a non-standard syntax" I can add it (such things aren't just for Solaris)... however I feel adding such comments everywhere would be a job for life ;-)

> Isn't shell check able to to check posix? But I guess that is syntax only and not sed/grep and others...

I suspect so yeah. There have also been cases where we have used `if` blocks to explicitly check for and use non-standard versions, so they'd have to be skipped.